interface/batchcom/batchcom.php

```diff
@@ -24,280 +29,217 @@
         // gender
-        if ($_POST['gender']!='Any') {
-            $and=where_or_and ($and);
-            $sql.=" $and patient_data.sex='".$_POST['gender']."' ";
-        }
-
+        if ($_POST['gender']!='Any') { $sql .= " and patient_data.sex='".$_POST['gender']."' "; }//INJECTION VECTOR HERE
         // hipaa overwrite
-        if ($_POST['hipaa_choice']!='NO') {
-            $and=where_or_and ($and);
-            $sql.=" $and patient_data.hipaa_mail='YES' ";
-        }
+        if ($_POST['hipaa_choice'] != $hipaa_choices[0]) { $sql .= " and patient_data.hipaa_mail='YES' "; }
 
         switch ($_POST['process_type']):
             case $choices[1]: // Email
-                $and=where_or_and ($and);
-                $sql.=" $and patient_data.email IS NOT NULL ";
+                $sql.=" and patient_data.email IS NOT NULL ";
             break;
         endswitch;
 
         // sort by
-        $sql.=' ORDER BY '.$_POST['sort_by'];
-        //echo $sql;
+        $sql.=' ORDER BY '.$_POST['sort_by'];//INJECTION VECTOR
         // send query for results.
+        //echo $sql;exit();
         $res = sqlStatement($sql);
 
-        // if no results.
         if (sqlNumRows($res)==0){
-	?>
-        <html>
-	<head>
-    <title><?php echo xlt('BatchCom'); ?></title>
-	<?php html_header_show();?>
-	<link rel="stylesheet" href="<?php echo $css_header;?>" type="text/css">
-	<link rel="stylesheet" href="batchcom.css" type="text/css">
-    <link rel="stylesheet" href="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-datetimepicker-2-5-4/build/jquery.datetimepicker.min.css">
-
-    <script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-min-3-1-1/index.js"></script>
-    <script type="text/javascript" src="<?php echo $GLOBALS['assets_static_relative']; ?>/jquery-datetimepicker-2-5-4/build/jquery.datetimepicker.full.min.js"></script>
-
-    <script LANGUAGE="JavaScript">
-    $(document).ready(function() {
-        $('.datepicker').datetimepicker({
-            <?php $datetimepicker_timepicker = false; ?>
-            <?php $datetimepicker_showseconds = false; ?>
-            <?php $datetimepicker_formatInput = false; ?>
-            <?php require($GLOBALS['srcdir'] . '/js/xl/jquery-datetimepicker-2-5-4.js.php'); ?>
-            <?php // can add any additional javascript settings to datetimepicker here; need to prepend first setting with a comma ?>
-        });
-    });
-    </script>
-
-	</head>
-	<body class="body_top">
-	<!-- larry's sms/email notification -->
-	<span class="title"><?php include_once("batch_navigation.php");?></span>
-	<!--- end of larry's insert -->
-	<span class="title"><?php xl('Batch Communication Tool','e')?></span>
-	<br><br>
-	<div class="text">
-        <?php
-            echo (xl('No results found, please try again.','','<br>'));
-        ?> </div></body></html> <?php
-        //if results
+            $form_err = xl('No results found, please try again.');
```

Comment:
To provide an example of when it's best to sometimes separate xl() and then use text/attr() separately. In this case, you are correctly just translating and not html escaping. The html escaping (with either attr() or text()) can then be done when this variable is echo'd to the screen.